src/download.py

```diff
@@ -69,12 +93,30 @@ def main():
             d.shift = timedelta(hours=args.shift)
             _log.info("Applying {:f} hours shift".format(args.shift))
         d.read(w, p)
+
+        if(len(w.tracks_written()) > 0 ):
+          table_rows = []
+          n = 1
+          for track in w.tracks_written():
+            table_rows.append([n, track['Track'], track['Start'], track['End'] - track['Start'], "%.1fkm/h" % (track['Speed']), track['Laps'], track['Points'], "%.2fkm" % (track['Distance'])])
+            n += 1
+          print(tabulate(table_rows, headers=["Track","Label","Started At","Duration","Speed","Laps","Points","Distance"]))
+            
+          if args.strava:
+            chosentracklist = raw_input("choose tracks to upload (eg. 1,3,4):")
```

Comment:
Good question.  I think I just wanted the option of choosing what to upload (as you can currently with the nasty windows software for Soleus watches).  But on reflection yes, why not just upload everything to strava and then use their UI to decide which ones you want to keep.  That would certainly make the code nicer.

Sorry, I didn't get as far as discovering post-processing scripts.  I was a bit conflicted about where to call the upload code, but that sounds like the place I was looking for...